joint_trajectory_controller/src/cartesian_trajectory_generator.cpp

```diff
@@ -273,6 +203,124 @@ void CartesianTrajectoryGenerator::reference_callback(
   add_new_trajectory_msg(new_traj_msg);
 }
 
+void CartesianTrajectoryGenerator::set_joint_limits_service_callback(
+  const std::shared_ptr<SetLimitsModeSrvType::Request> request,
+  std::shared_ptr<SetLimitsModeSrvType::Response> response)
+{
+  response->ok = true;
+  if (request->names.size() != request->limits.size())
+  {
+    RCLCPP_WARN(
+      get_node()->get_logger(),
+      "Fields name and limits have size %zu and %zu. Their size should be equal. Ignoring "
+      "service call!",
+      request->names.size(), request->limits.size());
+    response->ok = false;
+  }
+
+  // start with current limits
+  auto new_joint_limits = joint_limits_;
+
+  // lambda for setting new limit
+  auto update_limit_from_request = [](
+                                     double & new_limit_value, bool & new_limit_has_limits,
+                                     const double request_limit_value,
+                                     const double configured_limit_value)
+  {
+    // request is to reset to configured limit
+    if (std::isnan(request_limit_value))
+    {
+      new_limit_value = configured_limit_value;
+    }
+    // new limit is requested
+    else
+    {
+      new_limit_value = request_limit_value;
+    }
+    new_limit_has_limits = !std::isnan(new_limit_value);
```

Comment:
Shouldn't you check if there is a limit at all? What is this flag is set to `false` and what why are you setting this value?

Reply:
I'm afraid I don't understand the question. What limit should I be checking?

Reply:
Let's take a concrete example. If 'has_velocity_limits' is set to false, and we check it here, then we can never change velocity limits - no? 

I guess what you are saying is that if the configured limits has velocity limits disabled (has_velocity_limits = false), then we should never be changing velocity limits, since it is always disabled. Is that what you meant? Maybe, that is a valid assumption. In the same vein, if velocity limits is enabled in configured limits, then either we are changing the values through the service or going back to configured limits. I think this is OK.
Is this what you are meaning?

Reply:
I would say that if "has_velocity_limits" is set in service call then we have to change it, other wise we just ignore the value, i.e., keep the current value whatever that is

Reply:
Oh, but service message doesn't have a 'has_velocity_limits' flag. I was wondering why it didn't. Maybe we should change it.

Reply:
Oh, sorry, I thought it had. Then we can leave it as it is for now, and we will see if we need this when we start to use it. Or what do you think?

Reply:
I'll merge this for now because everything else is anyway a follow-up.